Conda setup for CI

rth opened the pull request:
Following discussion in https://github.com/pytest-dev/pytest-qt/issues/180  this PR proposes a Travis CI setup using conda. Please let me know if I missed anything.

**Edit:** with the appveyor.yml addition this fixes https://github.com/pytest-dev/pytest-qt/issues/180

nicoddemus commented on .travis.yml:
```diff
@@ -3,45 +3,48 @@ sudo: required
- - sudo $PYTHON -m pip install -U pytest tox coveralls
+ - |
+     if [[ ${PYTEST_QT_API} == 'pyside' ]]; then
+         pyqt_package="pyside==1.2.4"
```
What do you think about creating a `PYQT_PACKAGE` environment variable in the `env` section along with the others? There will be a small repetition, but I think that's OK.

nicoddemus replied:
I think we can use fuzzy versions like `pyside=1.*`, `pyqt=4.*` and `pyqt=5.*`.

nicoddemus commented:
Also please make sure to add yourself as a contributor in the `README`.

rth commented:
@nicoddemus Thanks a lot for the review! I addressed your comments and added the Appveyor setup (largely inspired by the one in qtpy). Also refactored a bit travis setup to be more consistent. Tests now pass. The only point is that in Travis there is a warning/error at the very end,
```
Done. Your build exited with 0.
XIO:  fatal IO error 11 (Resource temporarily unavailable) on X server ":99.0"
      after 2266 requests (2266 known processed) with 0 events remaining.
```
but I guess since tests pass that's not relevant...

rth commented:
Also there is no linting in Appveyor, but since it's alredy done in Travis that should, hopefully be sufficient..

nicoddemus commented on appveyor.yml:
```diff
@@ -1,27 +1,60 @@
+      CONDA_DEPENDENCIES: "pytest tox coveralls pyside=1.*"
+    - PYTHON_VERSION: "3.4"
+      PYTEST_QT_API: "pyqt5"
+      CONDA_DEPENDENCIES: "pytest tox coveralls pyqt=5.*"
```
We are not using `tox` and `coveralls` here so I guess we can remove them.

rth commented:
Thanks, you are right - I removed tox and coveralls from the Appveyor setup..

nicoddemus commented:
Awesome, thank you, really appreciate it! 👍
